okapi-core/src/main/java/org/folio/okapi/managers/KubernetesManager.java

```diff
@@ -0,0 +1,250 @@
+package org.folio.okapi.managers;
+
+import io.vertx.config.yaml.YamlProcessor;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.file.FileSystem;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.ConfNames;
+import org.folio.okapi.bean.DeploymentDescriptor;
+import org.folio.okapi.common.Config;
+import org.folio.okapi.common.ModuleId;
+import org.folio.okapi.common.OkapiLogger;
+
+public class KubernetesManager {
+
+  private static final Logger logger = OkapiLogger.get();
+
+  static final String KUBE_INST_PREFIX = "kube_";
+  final String fname;
+  int refreshInterval;
+  String token;
+  String server;
+  final String namespace;
+  WebClient webClient;
+  final WebClientOptions webClientOptions;
+  final DiscoveryManager discoveryManager;
+
+  /**
+   * Construct Kubernetes manager.
+   * @param config configuration.
+   */
+  public KubernetesManager(DiscoveryManager discoveryManager, JsonObject config) {
+    this.discoveryManager = discoveryManager;
+    refreshInterval = Config.getSysConfInteger(ConfNames.KUBE_REFRESH_INTERVAL, 30000, config);
+    fname = Config.getSysConf(ConfNames.KUBE_CONFIG, null, config);
+    token = Config.getSysConf(ConfNames.KUBE_TOKEN, null, config);
+    server = Config.getSysConf(ConfNames.KUBE_SERVER, null, config);
+    namespace = Config.getSysConf(ConfNames.KUBE_NAMESPACE, "default", config);
+    webClientOptions = new WebClientOptions()
+        .setTrustAll(Config.getSysConfBoolean(ConfNames.HTTP_CLIENT_TRUST_ALL, false, config));
```

Comment:
Never use setTrustAll(true).
This opens a back-door and defeats the security SSL/TLS is used for.

Example how to properly use https: https://github.com/folio-org/okapi/blob/v4.12.0/okapi-core/src/main/java/org/folio/okapi/service/impl/PostgresHandle.java#L70-L80

Supporting https might be split off into a separate pull request: https://issues.folio.org/browse/OKAPI-1071

Reply:
The Kubernetes API server *is* https based. So we need something we can do for https to work for us in "less secure" invironments such as developments. We can wait with the modules, but why would't that use the same strategy.

Reply:
setTrustAll no longer in use. with server PEM for the API server, that gets me going for local microk8s environment.

Reply:
Nice, thanks!